lib/shopify_transporter/pipeline/magento/product/top_level_variant_attributes.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+require 'shopify_transporter/pipeline/stage'
+require 'shopify_transporter/shopify'
+
+module ShopifyTransporter
+  module Pipeline
+    module Magento
+      module Product
+        class TopLevelVariantAttributes < Pipeline::Stage
+          def convert(hash, record)
+            accumulator = TopLevelVariantAttributesAccumulator.new(record['variants'])
+            accumulator.accumulate(hash)
+          end
+
+          class TopLevelVariantAttributesAccumulator < Shopify::AttributesAccumulator
+            COLUMN_MAPPING = {
+              'sku' => 'sku',
+              'weight' => 'grams',
+              'price' => 'price',
+            }
+            def input_applies?(input)
+              true unless input['parent_id'].nil?
+            end
+
+            def attributes_from(input)
+              map_from_key_to_val(COLUMN_MAPPING, input)
+            end
+
+            def accumulate_attributes(attributes)
```

Comment:
Why do we need the `delete_if`? Just curious.

Reply:
The function argument ```record``` comes in with this format: an key value pair with key being```variants``` and value being an array of simple products. The simple products are from the last pipeline stage so they are in Magento format. The delete_if is to get rid of the simple product in Magento format and the code in the next line will insert the product in Shopify format to the array

Reply:
I think we don't actually care about them enough to delete them as there's no side effect of their presence after-the-fact. As long as the attributes are moved to the right hash, all is good